floris/tools/optimization/scipy/yaw.py

```diff
@@ -216,6 +239,56 @@ def optimize(self, verbose=True):
 
         return opt_yaw_angles
 
+    def _reduce_control_variables(self):
+        """This function reduces the control problem by eliminating turbines
+        of which the yaw angles need not be optimized, either because of a
+        user-specified set of bounds (where bounds[i][0] == bounds[i][1]),
+        or alternatively turbines that are far downstream in the wind farm
+        and of which the wake does not impinge other turbines, if the
+        boolean exclude_downstream_turbines == True.
+        Secondly, the normalized initial conditions and bounds are calculated
+        for the subset of turbines, to be then used in the optimization
+        algorithm.
+        """
+        bnds = self.bnds
+        if self.bnds is not None:
+            self.turbs_to_opt = np.where(np.diff(self.bnds) > 0.001)[0]
+        else:
+            self.turbs_to_opt = np.array(range(self.nturbs), dtype=int)
+
+        if self.exclude_downstream_turbines:
+            downstream_turbines = derive_downstream_turbines(
+                fi=self.fi,
+                wind_direction=self.fi.floris.farm.wind_direction[0]
+            )
+            for i in downstream_turbines:
```

Comment:
Great point. I intended this to apply bounds to downstream turbines that have non-zero bounds. Basically, this would allow the user to assign a non-zero fixed yaw angle for these downstream machines. I am revising this by directly overwriting the initial conditions (self.x0) instead of the bounds. This should fix those issues.

Reply:
I might not quite understand, but if a user wants a certain fixed yaw angle for a given turbine, wouldn't that already be defined in self.x0? And couldn't overwriting self.x0 here cause some problems later since it might no longer be the value passed in to the object?

Reply:
Good points Eric. I think I've been making it unnecessarily complicated. I have now removed this entire part from `for i in downstream_turbines:`. Instead, I added a check in the optimizer initialization to make sure that x0 satisfies the bounds. if x0 does not satisfy the bounds, an error is returned to the user. x0 is now no longer automatically changed/updated within the code to avoid confusion.